qiskit/transpiler/passes/optimization/split_2q_unitaries.py

```diff
@@ -0,0 +1,49 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Splits each two-qubit gate in the `dag` into two single-qubit gates, if possible without error."""
+
+import numpy as np
+from qiskit.circuit.library import UnitaryGate
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.synthesis.two_qubit.local_invariance import two_qubit_local_invariants
+from qiskit.synthesis.two_qubit.two_qubit_decompose import decompose_two_qubit_product_gate
+from qiskit.transpiler import TransformationPass
+
+
+class Split2QUnitaries(TransformationPass):
+    """Splits each two-qubit gate in the `dag` into two single-qubit gates, if possible without error."""
+
+    def run(self, dag: DAGCircuit):
+        """Run the Split2QUnitaries pass on `dag`."""
+        for node in dag.topological_op_nodes():
+            # skip operations without two-qubits and for which we can not determine a potential 1q split
+            if (
+                len(node.cargs) > 0
```

Comment:
Do you think it makes sense to have a unitary gate name check here with:

```suggestion
                node.name != "unitary"
                or len(node.cargs) > 0
```

Or are you thinking there is a case where we have a another 2q gate type we can decompose into a product of 1q gates?

Reply:
I would suggest to no limit this pass to unitary gates, as this is more general and appears to work well. We can process conditional rotation gates with a zero gate here or look into any two-qubit gate that was not picked up by `ConsolidateBlocks` for any reason when not limiting to unitary gates only.